components/productCard/Splash.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <div v-if="splash" class="splash">
+    <div v-if="splash.label" class="promo-text">{{ splash.label }}</div>
+    <div class="price">
+      {{ splash.price }}<span class="unit">{{ splash.unit }}</span>
+      <span v-if="splash.decimal" class="decimal">{{ splash.decimal }}</span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Product from '~/types/Product'
+
+export default Vue.extend({
+  props: {
+    product: {
+      type: Object as () => Product,
+      required: true
+    }
+  },
+  computed: {
+    splash(): any {
+      if (!this.product.potentialPromotions[0]) {
+        return false
+      }
+      const regex = /(?<label>\d+ för )*(?<price>\d+):(?<decimal>[-\d]+) *\/*(?<unit>.*)/g
+      // https://regex101.com/
+
+      const parts = regex.exec(this.product.potentialPromotions[0].description)
+
+      if (!parts) {
+        window.console.error('Kan inte lista ut erbjudandet i splash')
+        return false
+      }
+
+      const label = parts[1]
+      let price = parts[2]
+      let decimal = parts[3]
+      let unit = parts[4]
+
+      if (unit === 'st' && decimal === '-') {
+        price = price + ':-'
+        decimal = ''
+        unit = ''
+      }
+
+      if (unit === '' && decimal === '-') {
+        price = price + ':-'
+        decimal = ''
+      }
+
+      if (unit !== '') {
+        unit = '/' + unit
+      }
+
+      return { label, price, decimal, unit }
+    },
+    splashText(): any {
```

Comment:
splashText, splashTextPrice, splashPrice and splashDecimal  is not used, you can delete these-